server/cmd/api/api/playwright_test.go

```diff
@@ -0,0 +1,94 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/onkernel/kernel-images/server/lib/logger"
+)
+
+func TestExecutePlaywrightRequest_Validation(t *testing.T) {
+	s := &Service{}
+
+	tests := []struct {
+		name           string
+		requestBody    string
+		expectedStatus int
+		checkError     bool
+	}{
+		{
+			name:           "empty code",
+			requestBody:    `{"code": ""}`,
+			expectedStatus: http.StatusBadRequest,
+			checkError:     true,
+		},
+		{
+			name:           "missing code field",
+			requestBody:    `{}`,
+			expectedStatus: http.StatusBadRequest,
+			checkError:     true,
+		},
+		{
+			name:           "invalid json",
+			requestBody:    `{invalid}`,
+			expectedStatus: http.StatusBadRequest,
+			checkError:     true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req := httptest.NewRequest(http.MethodPost, "/playwright/execute", bytes.NewBufferString(tt.requestBody))
+			req.Header.Set("Content-Type", "application/json")
+
+			testLogger := slog.New(slog.NewTextHandler(os.Stdout, nil))
+			ctx := logger.AddToContext(context.Background(), testLogger)
+			req = req.WithContext(ctx)
+
+			w := httptest.NewRecorder()
+
+			s.ExecutePlaywrightCode(w, req)
+
+			if w.Code != tt.expectedStatus {
+				t.Errorf("expected status %d, got %d", tt.expectedStatus, w.Code)
+			}
+		})
+	}
+}
+
+func TestExecutePlaywrightRequest_ValidCode(t *testing.T) {
```

Comment:
remove this test and add a new one to e2e_chromium_test.go that spins up the container and runs this api request against the running container, validating that the playwright script output is returned